File: 2023/12/06/fizzbuzz-refucktoring/src/fizzbuzz.ts
const toto = (tutu: number) => {
  let i = tutu
  while (i > 0) {
    i -= 15
    if (i === 0) {
      return true
    }
  }
  return false
}

export function fizzbuzz(number: number) {
  if (toto(number)) {
    return "Fizz Buzz"
  } else if (number % 3 === 0) {
    return "Fizz"
  } else if (number % 5 === 0) {
    return "Buzz"
  }
  return "" + number
}

File: 2025/10/17/lib/fizzbuzz.ex
defmodule Fizzbuzz do
  @moduledoc """
  Documentation for `Fizzbuzz`.
  """

  def fizzbuzz(n) do
    Stream.zip_with(
      [
        1..n,
        Stream.cycle([nil, nil, "fizz"]),
        Stream.cycle([nil, nil, nil, nil, "buzz"])
      ],
      fn [number, fizz, buzz] ->
        case {fizz, buzz} do
          {nil, nil} ->
            "#{number}"

          {fizz, nil} ->
            fizz

          {nil, buzz} ->
            buzz

          {fizz, buzz} ->
            fizz <> buzz
        end
      end
    )
    |> Enum.to_list()
  end
end
